### PR TITLE
Added reddit loader

### PR DESCRIPTION
**Pull request by siiddhantt**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

    Added the reddit loader from langchain (https://python.langchain.com/docs/integrations/document_loaders/reddit), as a 
    remote loader

### CODE REVIEW

File: application/parser/remote/reddit_loader.py [hunk 1]
```diff
@@ -4,24 +4,22 @@
...
+        mode = data.get("mode", "subreddit")
+        search_queries = data.get("search_queries")
         self.loader = RedditPostsLoader(
             client_id=client_id,
             client_secret=client_secret,
             user_agent=user_agent,
             categories=categories,
             mode=mode,
             search_queries=search_queries,
+            number_posts=10,
```
Comment (dartpain) [inline application/parser/remote/reddit_loader.py:21]: Lets make sure users have an option to pass this as a parameter

**Review by dartpain — CHANGES_REQUESTED** [review]
Just a tiny change that will allow users to pass number of posts as a parameter

File: application/parser/remote/reddit_loader.py (hunk 1 above)
Comment (siiddhantt) [reply application/parser/remote/reddit_loader.py]: sure, i've made the change for that now

**Review by dartpain — APPROVED** [review]
Good Job, Thank you!

**Pull request merged**